modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/secret/management/v1/SecretManagementFailureTest.java

```diff
@@ -0,0 +1,205 @@
+import org.wso2.carbon.utils.multitenancy.MultitenantConstants;
+
+import java.io.IOException;
+
+import static org.hamcrest.core.IsNull.notNullValue;
+import static org.testng.Assert.assertNotNull;
+
+/**
+ * Test class for Secret Management REST APIs failure paths.
+ */
+public class SecretManagementFailureTest extends SecretManagementTestBase {
+
+    private String secretTypeName = "sample-secret-type";
+    private String secretName = "sample-secret";
+
+    @Factory(dataProvider = "restAPIUserConfigProvider")
+    public SecretManagementFailureTest(TestUserMode userMode) throws Exception {
+
+        super.init(userMode);
+        this.context = isServer;
+        this.authenticatingUserName = context.getContextTenant().getTenantAdmin().getUserName();
+        this.authenticatingCredential = context.getContextTenant().getTenantAdmin().getPassword();
+        this.tenant = context.getContextTenant().getDomain();
+    }
+
+    @BeforeClass(alwaysRun = true)
+    public void init() throws IOException {
+
+        super.testInit(API_VERSION, swaggerDefinition, tenant);
+    }
+
+    @AfterClass(alwaysRun = true)
+    public void testConclude() {
+
+        super.conclude();
+    }
+
+    @BeforeMethod(alwaysRun = true)
+    public void testInit() {
+
+        RestAssured.basePath = basePath;
+    }
+
+    @AfterMethod(alwaysRun = true)
+    public void testFinish() {
+
+        RestAssured.basePath = StringUtils.EMPTY;
+    }
+
+    @DataProvider(name = "restAPIUserConfigProvider")
+    public static Object[][] restAPIUserConfigProvider() {
+
+        return new Object[][]{
+                {TestUserMode.SUPER_TENANT_ADMIN},
+                {TestUserMode.TENANT_ADMIN}
+        };
+    }
+
+    @Test
+    public void testAddSecretTypeConflict() throws IOException {
+
+        if (!StringUtils.equals(MultitenantConstants.SUPER_TENANT_DOMAIN_NAME, tenant)) {
+            String body = readResource("add-secret-type.json");
+            Response response =
+                    getResponseOfPost(SECRET_TYPE_API_BASE_PATH, body);
+
+            response.then()
+                    .log().ifValidationFails()
+                    .assertThat()
+                    .statusCode(HttpStatus.SC_CREATED)
+                    .header(HttpHeaders.LOCATION, notNullValue());
+            String location = response.getHeader(HttpHeaders.LOCATION);
+            assertNotNull(location);
+            secretTypeName = location.substring(location.lastIndexOf("/") + 1);
+            assertNotNull(secretTypeName);
+            response = getResponseOfPost(SECRET_TYPE_API_BASE_PATH, body);
+            validateErrorResponse(response, HttpStatus.SC_CONFLICT, "SECRETM_00019");
+        }
+    }
+
+    @Test
+    public void testGetSecretTypeByInvalidName() {
+
+        Response response = getResponseOfGet(SECRET_TYPE_API_BASE_PATH + PATH_SEPARATOR + "randomName");
+
+        validateErrorResponse(response, HttpStatus.SC_NOT_FOUND, "SECRETM_00021");
+
```

Comment:
Unnecessary newline

Reply:
Addressed with commit [252de16](https://github.com/wso2/product-is/pull/12546/commits/252de169987046a367e9c09f983731935058c944)